db/database.rb

```diff
@@ -68,9 +68,9 @@ class Organization
 
 class Repository
   include DataMapper::Resource
-  property :rname, String, :required => true, :key => true
+  property :rname, String, :required => true, :key => true, :unique_index => false
 
-  belongs_to :organization, :required => false
+  belongs_to :organization, :required => false, :key => true
```

Comment:
I fully agree to add this here. The only problem now is that I cannot migrate my existing DB to this. I will have to take a look how a migration could be done here.

```
`execute_non_query': Cannot add a NOT NULL column with default value NULL (DataObjects::SyntaxError)
```